File: src/main/java/xyz/chunkstories/core/converter/mappings/Chest.java
//
// This file is a part of the Chunk Stories Core codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.core.converter.mappings;

import xyz.chunkstories.api.converter.mappings.NonTrivialMapper;
import xyz.chunkstories.api.voxel.Voxel;
import xyz.chunkstories.api.world.World;
import xyz.chunkstories.api.world.cell.FutureCell;
import xyz.chunkstories.api.world.chunk.Chunk;
import io.xol.enklume.MinecraftRegion;

public class Chest extends NonTrivialMapper {

	public Chest(Voxel voxel) {
		super(voxel);
	}

	@Override
	public void output(World csWorld, int csX, int csY, int csZ, int minecraftBlockId, int minecraftMetaData,
			MinecraftRegion region, int minecraftCuurrentChunkXinsideRegion, int minecraftCuurrentChunkZinsideRegion,
			int x, int y, int z) {

		Chunk chunk = csWorld.getChunksManager().getChunkWorldCoordinates(csX, csY, csZ);
		assert chunk != null;

		csWorld.pokeSimpleSilently(new FutureCell(csWorld, csX, csY, csZ, voxel));
	}

}

File: src/main/java/xyz/chunkstories/core/converter/mappings/Chest.kt
//
// This file is a part of the Chunk Stories Core codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.core.converter.mappings

import io.xol.enklume.MinecraftRegion
import xyz.chunkstories.api.block.BlockType
import xyz.chunkstories.api.converter.NonTrivialMapper
import xyz.chunkstories.api.world.World

class Chest(blockType: BlockType) : NonTrivialMapper(blockType) {

    override fun output(csWorld: World, csX: Int, csY: Int, csZ: Int, minecraftBlockId: Int, minecraftMetaData: Int,
                        region: MinecraftRegion, minecraftCurrentChunkXinsideRegion: Int, minecraftCurrentChunkZinsideRegion: Int,
                        x: Int, y: Int, z: Int) {
        csWorld.getCellMut(csX, csY, csZ)!!.data.blockType = blockType
    }

}